vc_zoom/indico_vc_zoom/util.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2020 CERN and ENEA
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import random
+import string
+
+from requests.exceptions import HTTPError
+
+from indico.core.db import db
+from indico.modules.users.models.emails import UserEmail
+from indico.modules.users.models.users import User
+from indico.modules.vc.exceptions import VCRoomError, VCRoomNotFoundError
+from indico.util.date_time import now_utc
+
+from indico_vc_zoom import _
+from indico_vc_zoom.api import ZoomIndicoClient
+
+
+def find_enterprise_email(user):
+    """Find a user's first e-mail address which can be used by the Zoom API.
+
+    :param user: the `User` in question
+    :return: the e-mail address if it exists, otherwise `None`
+    """
+    from indico_vc_zoom.plugin import ZoomPlugin
+    providers = [auth.strip() for auth in ZoomPlugin.settings.get('email_domains').split(',')]
+    result = UserEmail.query.filter(
+        UserEmail.user == user,
+        ~User.is_blocked,
+        ~User.is_deleted,
+        db.or_(UserEmail.email.ilike("%%@{}".format(provider)) for provider in providers)
+    ).join(User).first()
```

Comment:
Should we order this query so a primary email comes first? Just in case someone has multiple emails with Zoom accounts for whatever reasons...